### PR TITLE
fix(console,phrases): improve error handling when user associated app is removed

### PR DESCRIPTION
**Pull request by charIeszhao**
## Summary

Improve error handling when the user associated app is removed.

## Testing

User management:
image

Audit logs:
image

Audit log details:
image

User logs:
image

## Checklist

- [x] `.changeset`
~~- [ ] unit tests~~
~~- [ ] integration tests~~
~~- [ ] necessary TSDoc comments~~

### CODE REVIEW

File: .changeset/silent-windows-behave.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@logto/console": patch
+"@logto/phrases": patch
+---
+
+Improve error handling when user associated application is removed.
```
Comment (gao-sun) [inline .changeset/silent-windows-behave.md:6]:
```suggestion
improve error handling when user associated application is removed
```

File: packages/console/src/components/ApplicationName/index.tsx [hunk 2]
```diff
@@ -16,13 +19,38 @@ type Props = {
...
+  const { data, error } = useSWR<Application, RequestError>(
+    !isAdminConsole && `api/applications/${applicationId}`,
+    {
+      fetcher,
+      shouldRetryOnError: (error: unknown) => {
+        if (error instanceof RequestError) {
+          return error.status !== 404;
+        }
+
+        return true;
```
Comment (gao-sun) [inline packages/console/src/components/ApplicationName/index.tsx:33]: i found similar code:

image

can we make a util for ignore certain status codes?

Comment (charIeszhao) [reply packages/console/src/components/ApplicationName/index.tsx:33]: ok

Comment (charIeszhao) [reply packages/console/src/components/ApplicationName/index.tsx]: Updated

File: packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Branding/use-application-sign-in-experience-swr.ts [hunk 3]
```diff
@@ -18,13 +19,7 @@ const useApplicationSignInExperienceSWR = (applicationId: string) => {
...
     {
       fetcher,
-      shouldRetryOnError: (error: unknown) => {
-        if (error instanceof RequestError) {
-          return error.status !== 404;
-        }
-
-        return true;
-      },
+      shouldRetryOnError: shouldRetryOnError({ ignore: [401, 403] }),
```
Comment (gao-sun) [inline packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Branding/use-application-sign-in-experience-swr.ts:22]: 404?

File: packages/console/src/utils/request.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { RequestError } from '@/hooks/use-api';
+
+type Options = {
+  ignore: number[];
+};
+
+/**
+ * Determines if a SWR request should be retried based on the error status code
+ * @param options (optional): `ignore` - an array of status codes to exclude from retrying
```
Comment (gao-sun) [inline packages/console/src/utils/request.ts:9]:
```suggestion
 * @param options - ...
 * @param options.ignore - an array of status codes to exclude from retrying
```

File: packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Branding/use-application-sign-in-experience-swr.ts (hunk 3 above)
Comment (charIeszhao) [reply packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/Branding/use-application-sign-in-experience-swr.ts:22]: Sh*t. My bad

**Review by gao-sun — APPROVED** [review]

**Pull request merged**